shared/desktop/remote/sync-props.desktop.js

```diff
@@ -5,69 +5,117 @@
   windowComponent: string,
   remoteWindow: ?SafeElectron.BrowserWindowType,
 }
 
-function SyncProps(ComposedComponent: any) {
-  class RemoteConnected extends React.PureComponent<Props> {
-    _lastProps: Object = {}
+// Before sending we need to get a small diff to serialize. Do NOT send immutable over the wire. You pass an object
+// for each matching prop which converts to a serializable format. Do smart diffing w/ the old value to minimize traffic
+type Serializer = {[key: string]: (value: any, oldValue: any) => ?Object}
+
+function SyncPropsFactory(serializer: Serializer) {
+  return function SyncProps(ComposedComponent: any) {
+    class RemoteConnected extends React.PureComponent<Props> {
+      _lastProps: any
 
-    _sendProps = () => {
-      if (this.props.remoteWindow) {
-        try {
-          const props = this._getPropsToSend()
-          // Using stringify to go over the wire as the representation it sends over IPC is very verbose and blows up
-          // the data a lot
-          this.props.remoteWindow && this.props.remoteWindow.emit('props', JSON.stringify(props))
-        } catch (e) {
-          console.error(e)
+      _sendProps = () => {
+        if (this.props.remoteWindow) {
+          try {
+            const props = this._getPropsToSend()
+            // Using stringify to go over the wire as the representation it sends over IPC is very verbose and blows up
+            // the data a lot
+            if (this.props.remoteWindow && Object.keys(props).length) {
+              this.props.remoteWindow.emit('props', JSON.stringify(props))
+            }
+          } catch (e) {
+            console.error(e)
+          }
         }
       }
-    }
 
-    _onNeedProps = ({sender}, windowComponent: string, windowParam: string) => {
-      if (windowComponent === this.props.windowComponent && windowParam === this.props.windowParam) {
-        // If the remote asks for props send the whole thing
-        this._lastProps = {}
-        this._sendProps()
+      _onNeedProps = ({sender}, windowComponent: string, windowParam: string) => {
+        if (windowComponent === this.props.windowComponent && windowParam === this.props.windowParam) {
+          // If the remote asks for props send the whole thing
+          this._lastProps = null
+          this._sendProps()
+        }
       }
-    }
 
-    _getPropsToSend = () => {
-      const childProps = this._getChildProps()
-      const toSend = Object.keys(childProps).reduce((map, key) => {
-        if (childProps[key] !== this._lastProps[key]) {
-          map[key] = childProps[key]
+      _getPropsToSend = () => {
+        const newProps = this._getChildProps()
+        const toSend = {}
+
+        // Do a shallow equal
+        if (debugSerializer) {
+          console.log('[Serializer]: ---------------------------------')
         }
-        return map
-      }, {})
-      this._lastProps = childProps
-      return toSend
-    }
+        Object.keys(newProps).forEach(k => {
```

Comment:
we do a shallow equal and only send different sub things through the serializer layer